items: port Item_Control to TRX

lahm86 opened the pull request:
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TRX/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change
- [x] I have added a readme entry about my new feature or OG bug fix, or it is a different change

#### Description

This shifts `Item_Control` to TRX, which was the last remaining items function. TR1 now has a minor difference compared with before, where control is not run on any killed items, but I don't think this was happening anyway from tests I've done. I'll do some extra general testing for this too, just opening it now to give plenty of time.

The function shift is the only logical change here, the rest is just related header tidy-up.

lahm86 committed:
items: port Item_Control to TRX

This moves the final remaining items function to TRX. Header includes
have been reorganised, plus TR1 now adopts the approach of not running
control against killed items.

rr- commented on src/tr1/game/objects/traps/sliding_pillar.c:
```diff
@@ -1,6 +1,14 @@
+typedef enum {
+    SPS_START = 0,
+    SPS_END = 1,
+    SPS_MOVING = 2,
```
This can likely be made private.

lahm86 replied:
Thanks, now renamed per convention.

lahm86 committed:
tr1/objects/sliding_pillar: rename state enum

This renames the animation state enum of the sliding pillar (naming
convention for private scope).
